elide-core/src/main/java/com/yahoo/elide/core/EntityBinding.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core;
+
+import com.yahoo.elide.annotation.Exclude;
+import com.yahoo.elide.core.exceptions.DuplicateMappingException;
+
+import lombok.Getter;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.text.WordUtils;
+
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedDeque;
+
+import javax.persistence.Column;
+import javax.persistence.Id;
+import javax.persistence.ManyToMany;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.OneToOne;
+import javax.persistence.PrePersist;
+import javax.persistence.Transient;
+
+/**
+ * Entity Dictionary Element Binding.  For use by EntityDictionary.
+ */
+class EntityBinding {
+    private final static List<Method> OBJ_METHODS = Arrays.asList(Object.class.getMethods());
+
+    public final String jsonApi;
+    public final ConcurrentLinkedDeque<String> attrsDeque;
+    public final List<String> attrs;
```

Comment:
Should we be using `public final` here? Though these fields are final, they're not all immutable. May not be a problem in practice, but we should think about it.